mbs_results/ratio_of_means.py

```diff
@@ -0,0 +1,388 @@
+from typing import Dict
+
+import numpy as np
+import pandas as pd
+
+from mbs_results.apply_imputation_link import create_and_merge_imputation_values
+from mbs_results.calculate_imputation_link import calculate_imputation_link
+from mbs_results.construction_matches import flag_construction_matches
+from mbs_results.cumulative_imputation_links import get_cumulative_links
+from mbs_results.flag_and_count_matched_pairs import count_matches, flag_matched_pair
+from mbs_results.imputation_flags import create_impute_flags, generate_imputation_marker
+from mbs_results.link_filter import flag_rows_to_ignore
+from mbs_results.predictive_variable import shift_by_strata_period
+
+
+def wrap_flag_matched_pairs(
+    df: pd.DataFrame, **default_columns: Dict[str, str]
+) -> pd.DataFrame:
+    """
+    Wrapper function for flagging forward, backward and construction pair
+    matches.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Original dataframe.
+    **default_columns : Dict[str, str]
+        The column names which were passed to ratio of means function.
+    Returns
+    -------
+    df : pd.DataFrame
+        Original dataframe with 4 bool columns, column names are
+        forward_or_backward keyword and target column name to distinguish them
+    """
+
+    if "ignore_from_link" in df.columns:
+
+        df["filtered_target"] = df["question"]
+        df.loc[df["ignore_from_link"], "filtered_target"] = np.nan
+        default_columns = {**default_columns, "target": "filtered_target"}
+
+    flag_arguments = [
+        dict(**default_columns, **{"forward_or_backward": "f"}),
+        dict(**default_columns, **{"forward_or_backward": "b"}),
+    ]
+
+    for args in flag_arguments:
+
+        df = flag_matched_pair(df, **args)
+
+    df = flag_construction_matches(df, **default_columns)
+
+    return df
+
+
+def wrap_count_matches(
+    df: pd.DataFrame, **default_columns: Dict[str, str]
+) -> pd.DataFrame:
+    """Wrapper function to get counts of flagged matched pairs.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Original dataframe.
+    **default_columns : Dict[str, str]
+        The column names which were passed to ratio of means function.
+    Returns
+    -------
+    df : pd.DataFrame
+        Dataframe with 3 numeric columns.
+    """
+
+    count_arguments = (
+        dict(**default_columns, **{"flag": "f_match"}),
+        dict(**default_columns, **{"flag": "b_match"}),
+        dict(**default_columns, **{"flag": "flag_construction_matches"}),
+    )
+
+    all_counts = pd.DataFrame(columns=default_columns.values())
+
+    for args in count_arguments:
+        counts = count_matches(df, **args)
+        all_counts = counts.merge(all_counts, how="left")
+
+    return all_counts
+
+
+def wrap_shift_by_strata_period(
```

Comment:
Same comments from `wrap_flag_matched_pairs` also apply to the following functions as they all use the same dictionary behaviour (again not bad just checking things might be overwritten)